### PR TITLE
V3 expose build strategy

### PR DESCRIPTION
**Pull request by jcantrill**

### CODE REVIEW

File: src/main/java/com/openshift3/client/model/build/ISTIBuildStrategy.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public interface ISTIBuildStrategy extends IBuildStrategy {
+	
+	/**
+	 * The Builder Image used to execute the build
+	 */
+	DockerImageURI getImage();
+	
+	String getScriptsLocation();
+	
+	List<EnviromentVariable> getEnvironmentVariables();
```
Comment (adietish) [inline src/main/java/com/openshift3/client/model/build/ISTIBuildStrategy.java]: not sure getting a list of environment variables is helpful. I would think that if you are interested in environment variables, you'd be interested in their value for a given key/name. Thus a map seems more helpful.

Comment (jcantrill) [inline src/main/java/com/openshift3/client/model/build/ISTIBuildStrategy.java]: @adietish per our discussion, converted to a map which makes more sense IMO

File: src/main/java/com/openshift3/internal/client/model/build/CustomBuildStrategy.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		this.env = env;
+	}
+	@Override
+	public BuildStrategyType getType() {
+		return BuildStrategyType.Custom;
+	}
+
+	@Override
+	public Map<String, String> getEnvironmentVariables() {
+		return env;
```
Comment (adietish) [inline src/main/java/com/openshift3/internal/client/model/build/CustomBuildStrategy.java]: I would not return the map as is, it's not protected from modifications by users.

File: src/main/java/com/openshift3/internal/client/model/build/DockerBuildStrategy.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	@Override
+	public String getContextDir() {
+		return contextDir;
+	}
+
+	@Override
+	public boolean isNoCache() {
+		// TODO Auto-generated method stub
```
Comment (adietish) [inline src/main/java/com/openshift3/internal/client/model/build/DockerBuildStrategy.java]: I guess that was left behind unintentionnally?

**Pull request merged**